### PR TITLE
feat(cli): leave numbers in dirname alone

### PR DESCRIPTION
**Pull request by hacksparrow**
Address https://github.com/strongloop/loopback-next/issues/6224.

We should simply lowercase the app name and use it as the dir name, `toFileName()` is best applicable to  "real" files only.

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/test/integration/generators/app.integration.js [hunk 1]
```diff
@@ -216,6 +216,45 @@ describe('app-generator with default values', () => {
   });
 });
 
+/** For testing if the app names with numbers are untouched */
+describe('app-generator with numbers in app name', () => {
+  const rootDir = path.join(__dirname, '../../../../../');
+  const defaultValProjPath = path.join(rootDir, 'sandbox/lb4-example');
+  const sandbox = path.join(rootDir, 'sandbox');
+  const pathToDefValApp = path.join(defaultValProjPath, 'lb4-example');
```
Comment (jannyHou) [inline packages/cli/test/integration/generators/app.integration.js:224]: so it means the project will be created inside 'sandbox/lb4-example/lb4-example'? Hmm why we have to use 2 lb4-example here?

Comment (hacksparrow) [reply packages/cli/test/integration/generators/app.integration.js:224]: Following the pattern in https://github.com/strongloop/loopback-next/blob/bc2293f70ad4cd6c450b3413a1302f133245837d/packages/cli/test/integration/generators/app.integration.js#L183.

**Review by raymondfeng — APPROVED** [review]

**Review by jannyHou — APPROVED** [review]
👍

**Comment by hacksparrow** [comment]
@bajtos any feedback on this?

**Comment by bajtos** [comment]
It makes me wonder how are we going to treat app names like `FooBar`? 

Inside loopback-next codebase, we have the convention to use CamelCase for identifiers like class names and kebab-case convention for file names. `PreferredCustomerModel` lives in `preferred-customer.model.ts` file.

I would expect that `lb4 app FooBar` creates the application in `foo-bar` directory with `FooBarApplication` as the app class name.

**Comment by hacksparrow** [comment]
@bajtos we cannot use uppercase in app names. Remember that `package.json` kinda rule? You will get `Generation is aborted: Invalid npm package name: FooBar`.

**Review by bajtos — APPROVED** [review]
> @bajtos we cannot use uppercase in app names. Remember that `package.json` kinda rule? You will get `Generation is aborted: Invalid npm package name: FooBar`.

Makes sense, I forgot that the name must be a valid npm package name. In that case it makes a lot of sense to me to use the raw package name as the directory name too 👍🏻

**Pull request merged**